tests/compile/test_sequence_parallelism.py

```diff
@@ -79,29 +79,95 @@ def ops_in_model(self):
         return [torch.ops._C.fused_add_rms_norm.default]
 
 
+class TestQuantModel(torch.nn.Module):
+
+    def __init__(self, hidden_size=16, intermediate_size=32):
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.gate_proj = torch.nn.Parameter(torch.empty(
+            (intermediate_size, hidden_size)),
+                                            requires_grad=False)
+        self.norm = RMSNorm(hidden_size, 1e-05)
+        # Initialize weights
+        torch.nn.init.normal_(self.gate_proj, std=0.02)
+        # Register scale as a buffer
+        self.register_buffer('scale', torch.tensor(1.0, dtype=torch.float32))
+
+    def forward(self, hidden_states, residual):
+        """
+        Forward pass implementing the operations in the FX graph
+        
+        Args:
+            hidden_states: Input tensor
+            residual: Residual tensor from previous layer
+            
+        Returns:
+            Tuple containing the output tensor
+        """
+        # Reshape input
+        view = hidden_states.reshape(-1, self.hidden_size)
+
+        #matrix multiplication
+        permute = self.gate_proj.permute(1, 0)
+        mm = torch.mm(view, permute)
+
+        # Tensor parallel all-reduce
+        all_reduce = tensor_model_parallel_all_reduce(mm)
+
+        # layer normalization
+        norm_output, residual_output = self.norm(all_reduce, residual)
+
+        scale = self.scale.to(norm_output.device)
+        quant_result = torch.empty(norm_output.shape,
+                                   dtype=current_platform.fp8_dtype())
+
+        torch.ops._C.static_scaled_fp8_quant.default(result=quant_result,
```

Comment:
Could you use `Fp8LinearOp` here to make sure if that interface changes this test catches it?